src/Domain/System/LogGateway.php

```diff
@@ -74,6 +74,39 @@ public function queryLogs(QueryCriteria $criteria, $gibbonSchoolYearID)
                     ->where('gibbonLog.gibbonPersonID = :gibbonPersonID')
                     ->bindValue('gibbonPersonID', $gibbonPersonID);
             },
+            'module' => function ($query, $module) {
+                return $query
+                    ->where('gibbonModule.name = :module')
+                    ->bindValue('module', $module);
+            },
+            'startDate' => function ($query, $startDate) {
+                return $query
+                    ->where('timestamp >= :startDate')
+                    ->bindValue('startDate', $startDate);
+            },
+            'endDate' => function ($query, $endDate) {
+                return $query
+                    ->where('timestamp <= :endDate')
+                    ->bindValue('endDate', $endDate);
+            },
+            'array' => function ($query, $array) {
+                $array = unserialize($array);
+                if (is_array($array)) {
+                    $count = 0;
+                    foreach ($array as $key => $value) {
+                        $bindKey = 'key' . $count;
+                        $bindValue = 'value' . $count;
+
+                        $query
+                            ->where('serialisedArray LIKE CONCAT("%", :'.$bindKey.', "%;%", :'.$bindValue.', "%")')
```

Comment:
Looks like some useful changes. I'm curious about the use case for the array filter. Is there a place in the codebase where this is needed, or is it hypothetical?

I'm looking at the LIKE syntax and I wonder if the % wildcards are too open ended, having a % between the key and the value means that any number of other values could come in the middle, so it may match a value for the wrong key. Perhaps a REGEXP may be more accurate here.

Reply:
As far as I can tell, its only ever used in the Help Desk when gathering logs from a specific technician, which is stored in the serialized array. I'll try experimenting with REGEXP.

Reply:
Okay, if its only ever used in that one spot, I'll merge it now and leave it to you to refine the filter later if you run into any issues there. Thanks!